### ISSUE TITLE
Add class API for metrics

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#497]
**Is your feature request related to a problem? Please describe.**
If we have class API for metrics, it makes easier to set all training components before training loop.
Get 3 feedback from Andriy:
1. Add class API for the metrics functions.
2. Change `torch.tensor(float(“nan”)).to(y_o.float())` to
`torch.tensor(float("nan"), device=y_o.device)`
3. Add `dim=` to several PyTorch calls.

**Comment by Nic-Ma** [comment]
Mark: @myron is working on a PR to resolve this issue.

**Comment by myron** [comment]
So, here is my attempt 
https://github.com/Project-MONAI/MONAI/pull/574

plz check

**Comment by wyli** [comment]
fixed by https://github.com/Project-MONAI/MONAI/pull/574
